### PR TITLE
Allow to parse more backend kwargs to pydap backend

### LINKED ISSUE #6274: Enable to parse more backend_kwargs when opening via PydapDataStore (e.g. output_grid)
### Is your feature request related to a problem?

I am planning to use xarray to access netcdf data from a thredds server. For my particular application, I need to be able to set pydap client option `output_grid=False`, to avoid some issues that I have experienced with a particular dataset on the server I am downloading from. 

I use `xarray.open_dataset` to access the data via pydap backend, which currently only allows to specify "session" as a keyword in `backend_kwargs`. 

https://github.com/pydata/xarray/blob/8c5c230881d616af32c5a42cca261eb11fce3916/xarray/backends/pydap_.py#L89

It would be helpful to be able to provide additional `backend_kwargs` accepted by the pydap client (such as `output_grid`).

https://github.com/pydap/pydap/blob/dbe9b8d15831f21c28b97ed5661b2708faec45f2/src/pydap/client.py#L58

### Describe the solution you'd like

It could be perhaps as simple as extending the signature of [PydapDataStore.open](https://github.com/pydata/xarray/blob/8c5c230881d616af32c5a42cca261eb11fce3916/xarray/backends/pydap_.py#L89) from:

```python
@classmethod
def open(cls, url, session=None):

    ds = pydap.client.open_url(url, session=session)
    return cls(ds)
```

to, e.g.:

```python
@classmethod
def open(cls, url, **kwargs):

    ds = pydap.client.open_url(url, **kwargs)
    return cls(ds)
```

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jgliss**
- [x] Closes #6274
- [x] Tests added
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`
- [x] New functions/methods are listed in `api.rst`

### CODE REVIEW

**Comment by jgliss** [comment]
I am not sure if the proposed changes are in compliance with the plugin restrictions related to parsing of **kwargs, etc. which is likely for very good reason. So, we could also be more explicit regarding how to deal with [pydap's defaults](https://github.com/pydap/pydap/blob/dbe9b8d15831f21c28b97ed5661b2708faec45f2/src/pydap/client.py#L58), directly in [PydapBackendEntrypoint](https://github.com/pydata/xarray/blob/8c5c230881d616af32c5a42cca261eb11fce3916/xarray/backends/pydap_.py#L116), and set the defaults directly (which could lead to unnoticed deviations from pydap's defaults in upcoming releases).

I appreciate comments and suggestions.

**Comment by mathause** [comment]
Looks like you need to define a `open_dataset_parameters` class variable if you want to pass on `kwargs`. I would prefer this but I don't have much experience with the backends, so others opinions may differ @aurghs @alexamici

See for example:

https://github.com/pydata/xarray/blob/8c5c230881d616af32c5a42cca261eb11fce3916/xarray/backends/pseudonetcdf_.py#L107-L111


If you want to keep the named arguments I would probably avoid `inspect` and just pass kwargs that are not None:

```python
kwargs = {"a": 5, "b": None}

# only pass on kwargs that are not None -> pydap has some defaults that are not None
kwargs = {key: value for key, value in kwargs.items() if value is not None}
```

**Comment by jgliss** [comment]
> Looks like you need to define a `open_dataset_parameters` class variable if you want to pass on `kwargs`. I would prefer this but I don't have much experience with the backends, so others opinions may differ @aurghs @alexamici
> 
> See for example:
> 
> https://github.com/pydata/xarray/blob/8c5c230881d616af32c5a42cca261eb11fce3916/xarray/backends/pseudonetcdf_.py#L107-L111
> 
> If you want to keep the named arguments I would probably avoid `inspect` and just pass kwargs that are not None:
> 
> ```python
> kwargs = {"a": 5, "b": None}
> 
> # only pass on kwargs that are not None -> pydap has some defaults that are not None
> kwargs = {key: value for key, value in kwargs.items() if value is not None}
> ```

@mathause thanks for your comments, I implemented your suggestion using `open_dataset_parameters`, hope this is sufficient. The code is covered by existing tests (using formerly keyword arg `session`), but let me know if you think I should add more tests, for the additional arguments.

File: xarray/backends/pydap_.py [hunk 1]
```diff
@@ -109,6 +108,18 @@ def get_dimensions(self):
 
 class PydapBackendEntrypoint(BackendEntrypoint):
     available = has_pydap
+    # *args and **kwargs are not allowed in open_backend_dataset_ kwargs,
```
Comment (mathause) [inline xarray/backends/pydap_.py:111]:
```suggestion

    # *args and **kwargs are not allowed in open_backend_dataset_ kwargs,
```

**Review by mathause — COMMENTED** [review]
LGTM - I hope someone with more backend experience finds time to take a look.

**Comment by aurghs** [comment]
I would prefer to avoid using **kwargs, the explicit list of parameters would make the code more readable. 
But I think it's fine that way too.

**Comment by jgliss** [comment]
> I would prefer to avoid using **kwargs, the explicit list of parameters would make the code more readable. But I think it's fine that way too :)

Thanks, rethinking it, I agree that it is better to set the allowed arguments explicitly. I updated it accordingly and hope this looks fine.

**Pull request merged**